### ISSUE TITLE
problem running this inside docker image

### ISSUE THREAD

**Issue by billtlee** [creativetimofficial/material-dashboard-react#84]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [x ] I am running the latest version
- [ x] I checked the documentation and found no answer
- [ x] I checked to make sure that this issue has not already been filed
- [ ] I'm reporting the issue to the correct repository (for multi-repository projects)

# Expected Behavior

I am working on an existing project within a docker image and wanted to use this as the admin frontend.  I gitcloned the directory and created the following docker file

FROM node:alpine
WORKDIR /app
COPY ./package.json ./
RUN npm install --g cross-env
RUN npm install
COPY . .
CMD ["npm", "run", "start"]

I also create a .env  with the NODE_PATH as well as modified the start script to include the NODE_PATH

However, I am still getting a module not found error on @material-ui/core/AppBar

# Current Behavior

getting module not found error on @material-ui/core/AppBar

# Failure Information (for bugs)

./src/components/Navbars/Navbar.jsx
Module not found: Can't resolve '@material-ui/core/AppBar' in '/app/src/components/Navbars'

## Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. git clone project into a directory
2. create dockerfile in the project directory (above content)
3. modify package.json to include NODE_PATH
4. create docker-compose.yml in parent project directory. The project is under client directory, so the last section
5. run docker-compose up --build from parent project directory

version: '3'
services: 
  redis:
    image: 'redis:latest'
  nginx:
      restart: always
      build:
        dockerfile: Dockerfile
        context: ./nginx
      ports:
        - '3000:80'
  api:
    build:
      dockerfile: Dockerfile
      context: ./server
    volumes:
      - /app/node_modules
      - ./server:/app
    environment:
      - REDIS_HOST=redis
      - REDIS_PORT=6379
      - EXPRESS_PORT=5000
  client:
    build:
      dockerfile: Dockerfile
      context: ./client
    volumes:
      - /app/node_modules
      - ./client:/app

**Comment by EINazare** [comment]
Hello there @billtlee ,

Thank you for your interest in working with our product.
Can you please paste your package.json here?

Best,
Manu

**Comment by billtlee** [comment]
{
  "name": "material-dashboard-react",
  "version": "1.6.0",
  "description": "Material Dashboard React. Coded by Creative Tim",
  "private": false,
  "main": "dist/index.js",
  "dependencies": {
    "@material-ui/core": "^3.9.2",
    "@material-ui/icons": "3.0.2",
    "chartist": "0.10.1",
    "classnames": "2.2.6",
    "history": "4.7.2",
    "perfect-scrollbar": "1.4.0",
    "prettier": "1.16.4",
    "prop-types": "15.7.1",
    "react": "16.8.1",
    "react-chartist": "0.13.3",
    "react-dom": "16.8.1",
    "react-google-maps": "9.4.5",
    "react-router-dom": "4.3.1",
    "react-scripts": "2.1.5",
    "react-swipeable-views": "0.13.1"
  },
  "scripts": {
    "start": "NODE_PATH=./src react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject",
    "install:clean": "rm -rf node_modules/ && rm -rf package-lock.json && npm install && npm start",
    "lint:check": "eslint . --ext=js,jsx;  exit 0",
    "lint:fix": "eslint . --ext=js,jsx --fix;  exit 0",
    "build-package-css": "cp src/assets/css/material-dashboard-react.css dist/material-dashboard-react.css",
    "build-package": "npm run build-package-css && babel src --out-dir dist"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/creativetimofficial/material-dashboard-react.git"
  },
  "keywords": [],
  "author": "Creative Tim <hello@creative-tim.com> (https://www.creative-tim.com/)",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/creativetimofficial/material-dashboard-react/issues"
  },
  "homepage": "https://creativetimofficial.github.io/material-dashboard-react/#/dashboard",
  "optionalDependencies": {
    "@types/googlemaps": "3.30.16",
    "@types/markerclustererplus": "2.1.33",
    "ajv": "6.9.1"
  },
  "browserslist": [
    ">0.2%",
    "not dead",
    "not ie <= 11",
    "not op_mini all"
  ],
  "devDependencies": {
    "eslint-config-prettier": "4.0.0",
    "eslint-plugin-prettier": "3.0.1"
  }
}

**Comment by billtlee** [comment]
I got it to work.  I specified the node version node:8.15-alpine, then I had to go into the container shell and manually install the packages and run npm audit fix to get it to work for some reason...  If someone can figure out why that'd be very helpful.  For the moment, I can continue to play with this.

**Comment by antondomratchev** [comment]
@billtlee its been a while since I've ran a react app in docker. I would double check that your node module are installing into `- /app/node_modules` and that your app can find that directory. Last time I tried this, i had to install node modules into the same volume as the react app. Again, I don't have a lot of knowledge with dock config, its something that jumped out at me from the error you are seeing, specifically, its looking for node_modules in this directory `/app/src/components/Navbars`
